### DISCUSSION TITLE
@types/react-table `FilterTypes<D>` typing is wrong

### DISCUSSION THREAD

**Discussion by akomm** [DefinitelyTyped/DefinitelyTyped#55920] [Issues with a @types package]
Version @types/react-table@7.7.4

Current, wrong typing (notice the generic D is not even used):
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6aafb9b17a6395742e81e49baced05add335c155/types/react-table/index.d.ts#L477

Probably was intended to be:
```ts
export type FilterTypes<D extends object> = Record<string, FilterType<D>>;
```
@ggascoigne, @stramel, @gargroh, @riceboyler

[a comment by an automated account is not shown]

**Comment by ggascoigne** [comment]
agreed, I think that you're correct.

**Comment by ggascoigne** [comment]
If you wouldn't mind posting a PR, it'll be much faster than me doing so.  If you post it, then I can approve it, if I post it, it'll likely wait a week.

**Comment by akomm** [comment]
Oh. I actually intended. But I never did this on the DT repo. Need to check if there is anything special about it then.

**Comment by akomm** [comment]
Going to provide a PR, even though I'v decided not to use react-table. Would feel guilty starting the report here and running away ;P

Not a blame on you guys, just a small vent on react-table:
I'm struggling to understand how one can not use any typing system officially AND not tell anything about a field, that is more than just a simple value:
https://react-table.tanstack.com/docs/api/useFilters#table-options
```
Must be memoized
Allows overriding or adding additional filter types for columns to use. If a column's filter type isn't found on this object, the column will default to using the built-in filter types.
```
How can one throw that in the face of people? You can `guess` it out from the examples, but still vague if you don't know the library. I understand when a library is not initial designed with typescript in mind, but when you don't have typing, you document?

**Comment by akomm** [comment]
When working on the fix I'v realized you have typing usage in readme. I actually never read `@types` readmes, because usually package maintainers add typescript docs on the official page if there are any. And if not, they document usage in a semi-ts way so it fits the typescript types.

Did not need it, because you'v typed and named things in an intuitive way. So I could find the types not even looking into the d.ts, just from the import autocomplete blindly-guessing type names are similar to hooks:
https://github.com/tannerlinsley/react-table/discussions/2567

It is not complete, just an example and not entirelly same semantics (type aliases/intersection), but the idea was clear without see your readme, how you probably intended to use it and also why that is (js library API is hard to auto-infer from the passed hooks).

Just wanted to give you all :+1: on that

**Reply by ggascoigne** [reply]
Thank you.  I really should talk to Tanner about putting the type docs on the main page, back when the typescript stuff moved out of the main package and over to DT it sort of made sense, but I suspect that now a useful pointer (or copy) would make more sense.

That said, I'm glad that you found the usage intuitive, that's really good to hear.
